### ISSUE TITLE
Category affiliation: If a category affiliation gets changed from series to video in the 'other' event labels end up showing up mutliple times

### ISSUE THREAD

**Issue by dagraf** [opencast/annotation-tool#601]
Steps to reproduce:
1. Have a series with at least two videos (A and B).
2. In video A: Create a category with at leas one label and set its affiliation to "Belonging to series".
3. In video B: Edit this category's affiliation to "Belongig to series". Click on save.
4. Still in video B: See that the labels get shown multiple times.
5. Still in video B: After a reload of the browser the labels do not get shown multiple times. But when opening the "Edit category" window the labels still show up multiple times.

After step 2 (see category "S - Teacher1"):
<img width="629" alt="Bildschirm­foto 2023-02-20 um 21 57 20" src="https://user-images.githubusercontent.com/15246287/220197394-28d2496e-0218-45e9-a4fa-2e7c1ea7958a.png">

After step 4:
<img width="641" alt="Bildschirm­foto 2023-02-20 um 21 58 10" src="https://user-images.githubusercontent.com/15246287/220197492-07dc3c48-0979-4ba4-b9e1-ba9ba28ada33.png">
<img width="637" alt="Bildschirm­foto 2023-02-20 um 21 58 23" src="https://user-images.githubusercontent.com/15246287/220197524-7c2a7636-9414-4054-8c03-6052636eba9c.png">

After step 5:
<img width="644" alt="Bildschirm­foto 2023-02-20 um 21 58 41" src="https://user-images.githubusercontent.com/15246287/220197554-601ab85a-a94b-4bcb-81af-5f745ef933f7.png">
<img width="625" alt="Bildschirm­foto 2023-02-20 um 21 58 55" src="https://user-images.githubusercontent.com/15246287/220197575-ee7eb3a7-5144-4e57-959f-d0cfdbfa9dea.png">

**Comment by JulianKniephoff** [comment]
So yes, this is very likely caused by the same bug as #645. This is a bigger problem, so I'll close this, and likely the other as well, and replace them with a new issue that captures the full extent of the problem.
